syntax = "proto3";

option java_multiple_files = true;
option java_package = "io.grpc.examples.helloworld";
option java_outer_classname = "HelloWorldProto";
option objc_class_prefix = "HLW";

package helloworld;

// The greeting service definition.
service Greeter {
  // Sends a greeting
  rpc ServiceCall (HelloRequest) returns (HelloReply) {}
}

// The request message containing the user's name.
message HelloRequest {
  enum Action{
    HelloWorld = 0;
  }
  Action action = 1;
  optional string name = 2;
}

// The response message containing the greetings
message HelloReply {
  enum ErrorCode {
    SUCCESS = 0;
    Fail = 1;
  }
  optional ErrorCode status = 1;
  optional string result = 2;
  optional uint32 err_code = 3;
  optional string err_msg = 4;
}
